internal/pkg/config/config.go

```diff
@@ -295,12 +306,29 @@ func (fm funcMatcher) MatchFunction(path, receiver, name string) bool {
 	return fm.Package.MatchString(path) && fm.Receiver.MatchString(receiver) && fm.Method.MatchString(name)
 }
 
-// ReadConfig fetches configuration from the config cache.
-// The cache reads, parses, and validates config file if necessary.
+// ReadConfig reads configuration from the config cache.
+// The cache reads, parses, and validates the config file if necessary.
+// If the config bytes were set using SetConfigBytes, they are used instead.
 func ReadConfig() (*Config, error) {
+	if configBytes != nil {
+		return readConfigBytes()
+	}
+
 	return cache.read(configFile)
 }
 
+func readConfigBytes() (*Config, error) {
+	if configFromBytes == nil {
+		configFromBytes = new(Config)
```

Comment:
It's been a while since I poked at the particulars of the Go memory model, but this seems thread-unsafe.  `configFromBytes` should either be `nil` or the final value to avoid instability.  Use an intermediate value for marshalling.  I'm assuming reassignment to the same value is thread safe.

Alternatively, use locks and a `cache` pattern similar to the read-from-file.

Reply:
Good catch. I think in this case an intermediate value + a comment makes the most sense. Done.